action/action.go

```diff
@@ -29,6 +30,90 @@ type Action interface {
 	Run(*claim.Claim, credentials.Set, io.Writer) error
 }
 
+func golangTypeToJSONType(value interface{}) (string, error) {
+	switch v := value.(type) {
+	case nil:
+		return "null", nil
+	case bool:
+		return "boolean", nil
+	case float64:
+		// All numeric values are parsed by JSON into float64s. When a value could be an integer, it could also be a number, so give the more specific answer.
+		if math.Trunc(v) == v {
+			return "integer", nil
+		}
+		return "number", nil
+	case string:
+		return "string", nil
+	case map[string]interface{}:
+		return "object", nil
+	case []interface{}:
+		return "array", nil
+	default:
+		return fmt.Sprintf("%T", value), fmt.Errorf("unsupported type: %T", value)
+	}
+}
+
+func setOutputsOnClaim(claim *claim.Claim, outputs map[string]string) error {
+	outputErrors := []string{}
+	claim.Outputs = map[string]interface{}{}
+
+	if claim.Bundle.Outputs != nil {
+		for outputName, v := range claim.Bundle.Outputs.Fields {
+			name := claim.Bundle.Outputs.Fields[outputName].Definition
+			outputSchema := claim.Bundle.Definitions[name]
+
+			var outputTypes []string
```

Comment:
nit: Can you extract this `outputTypes` and `mapOutputTypes` resolution in a sub function for better readability?